nats.go

```diff
@@ -4410,6 +4412,49 @@ func (s *Subscription) Drain() error {
 	return conn.unsubscribe(s, 0, true)
 }
 
+// IsDraining returns a boolean indicating whether the subscription
+// is being drained.
+// This will return false if the subscription has already been closed.
+// For blocking until the subscription is drained, use
+// [Subscription.DrainingComplete].
+func (s *Subscription) IsDraining() bool {
+	if s == nil {
+		return false
+	}
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.draining
+}
+
+// DrainingComplete returns a channel that will be closed when the
+// subscription is fully drained. If the subscription is nil, a closed
+// channel will be returned to avoid blocking on the caller side.
+func (s *Subscription) DrainingComplete() <-chan struct{} {
+	var drainCh chan struct{}
+	if s == nil {
+		// if the subscription is nil, return a closed channel
+		// to avoid blocking on the caller side.
+		drainCh = make(chan struct{})
+		close(drainCh)
+		return drainCh
+	}
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.drainCh == nil {
+		drainCh = make(chan struct{})
+		s.drainCh = drainCh
+	} else {
+		drainCh = s.drainCh
+	}
+	if !s.draining {
```

Comment:
I don't like that part. It can be misleading if someone is concurrently Draining and checking if drain is Drain is complete.
I know it should not be done. I just don't like that the result can mean almost opposit things.